src/dist/replication/lib/backup/replica_backup_manager.cpp

```diff
@@ -0,0 +1,130 @@
+    if (find != _replica->_cold_backup_contexts.end()) {
+        cold_backup_context_ptr backup_context = find->second;
+        if (backup_context->is_checkpointing()) {
+            ddebug_replica(
+                "{}: delay clearing obsoleted cold backup context, cause backup_status == "
+                "ColdBackupCheckpointing",
+                backup_context->name);
+            tasking::enqueue(LPC_REPLICATION_COLD_BACKUP,
+                             &_replica->_tracker,
+                             [this, request]() {
+                                 backup_response response;
+                                 on_cold_backup_clear(request);
+                             },
+                             get_gpid().thread_hash(),
+                             std::chrono::seconds(100));
+            return;
+        }
+
+        _replica->_cold_backup_contexts.erase(request.policy_name);
+    }
+
+    background_clear_backup_checkpoint(request.policy_name);
+}
+
+void replica_backup_manager::background_clear_backup_checkpoint(const std::string &policy_name)
+{
+    ddebug_replica("schedule to clear all checkpoint dirs of policy({}) in {} minutes",
+                   policy_name,
+                   _replica->options()->cold_backup_checkpoint_reserve_minutes);
+    tasking::enqueue(
+        LPC_BACKGROUND_COLD_BACKUP,
+        &_replica->_tracker,
+        [this, policy_name]() { clear_backup_checkpoint(policy_name); },
+        get_gpid().thread_hash(),
+        std::chrono::minutes(_replica->options()->cold_backup_checkpoint_reserve_minutes));
+}
+
+// clear all checkpoint dirs of the policy
+void replica_backup_manager::clear_backup_checkpoint(const std::string &policy_name)
+{
+    ddebug_replica("clear all checkpoint dirs of policy({})", policy_name);
+    auto backup_dir = _replica->_app->backup_dir();
+    if (!utils::filesystem::directory_exists(backup_dir)) {
+        return;
+    }
+
+    // Find the corresponding checkpoint dirs with policy name
+    std::vector<std::string> chkpt_dirs;
+    if (!get_policy_checkpoint_dirs(backup_dir, policy_name, chkpt_dirs)) {
+        dwarn_replica("get checkpoint dirs in backup dir({}) failed", backup_dir);
+        return;
+    }
+
+    // Remove these checkpoint dirs
+    for (const std::string &dirname : chkpt_dirs) {
+        std::string full_path = utils::filesystem::path_combine(backup_dir, dirname);
+        if (utils::filesystem::remove_path(full_path)) {
+            ddebug_replica("remove backup checkpoint dir({}) succeed", full_path);
+        } else {
+            dwarn_replica("remove backup checkpoint dir({}) failed", full_path);
+        }
+    }
+}
+
+void replica_backup_manager::send_clear_request_to_secondaries(const gpid &pid,
+                                                               const std::string &policy_name)
+{
+    backup_clear_request request;
+    request.__set_pid(pid);
+    request.__set_policy_name(policy_name);
```

Comment:
You can use `request.pid = pid`, filed `pid` in `backup_clear_request` is not optional.

Reply:
I think __set_pid is better, because it is more complicable